#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.5078284582252609 0.7666292175461505 0.3929248017383625 1.5270226831512388
  position 2.981035920311066 4.214337295135056 0.6837482301846899
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 3 3
  wallHeight 0.2
}
DEF WALL Solid {
  translation 0.887973 0 -0.578725
  rotation 0 1 0 -1.3089953071795861
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1.3 0.4 0.2
      }
    }
  ]
}
DEF WALL Solid {
  translation -0.325875 0 0.762907
  rotation 0 -1 0 -2.6179953071795863
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1.3 0.4 0.2
      }
    }
  ]
  name "solid(7)"
}
DEF WALL Solid {
  translation -0.287402 0 0.018239
  rotation 0 1 0 -2.094395307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 0.5 0.4 0.3
      }
    }
  ]
  name "solid(1)"
}
DEF WALL Solid {
  translation -0.00663559 0 -1.47521
  rotation 0 1 0 -2.8797953071795863
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1.1 0.4 0.37
      }
    }
  ]
  name "solid(9)"
}
DEF WALL Solid {
  translation -0.956474 0 -0.612519
  rotation 0 1 0 1.83259
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 0.6 0.4 0.1
      }
    }
  ]
  name "solid(8)"
}
DEF WALL Solid {
  translation 0.553039 0 0.548155
  rotation 0 1 0 2.35619
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1.1 0.4 0.15
      }
    }
  ]
  name "solid(2)"
}
DEF WALL Solid {
  translation -1.21468 0 -1.20543
  rotation 0 -1 0 -0.785395307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1 0.4 0.15
      }
    }
  ]
  name "solid(3)"
}
DEF WALL Solid {
  translation 1.19647 0 1.20573
  rotation 0 -1 0 -0.785395307179586
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1 0.4 0.15
      }
    }
  ]
  name "solid(4)"
}
DEF WALL Solid {
  translation -1.19353 0 1.20573
  rotation 0 -1 0 0.785406
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1 0.4 0.15
      }
    }
  ]
  name "solid(5)"
}
DEF WALL Solid {
  translation 1.23885 0 -1.22664
  rotation 0 -1 0 0.785406
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0.5 0
        metalness 0
      }
      geometry Box {
        size 1 0.4 0.15
      }
    }
  ]
  name "solid(6)"
}
DEF WheelBot Robot {
  translation 1.403195620919955 -0.00019880869073132107 -0.3289991142808627
  rotation 0.00035175574851501634 0.9995787742942476 0.02901982509084016 -3.111778426466737
  children [
    DEF WHEELBOTBODYSHAPE Transform {
      translation 0 0.04 0
      rotation 1 0 0 0
      children [
        Shape {
          appearance DEF BODYCOLOUR PBRAppearance {
            baseColor 0.4 0 0.3
            metalness 0
            IBLStrength 0.2
          }
          geometry DEF WHEELBOTBODY Cylinder {
            height 0.075
            radius 0.04
          }
        }
      ]
    }
    DEF COMPASS Group {
      children [
        Transform {
          translation 0 0.0775 0.011
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.7 0 0
                metalness 0
              }
              geometry Cone {
                bottomRadius 0.016
                height 0.05
                subdivision 14
              }
            }
          ]
        }
        Transform {
          translation 0 0.0775 -0.02
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                metalness 0
              }
              geometry Capsule {
                height 0.001
                radius 0.017
              }
            }
          ]
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -16130.99594321337
        axis -1 0 0
        anchor 0 0.02 0
      }
      device [
        RotationalMotor {
          name "leftmotor"
        }
      ]
      endPoint DEF WHEELBOTWHEEL Solid {
        translation 6.299322153416583e-09 0.029381962852418383 -0.017654176839754298
        rotation -0.9999999999999829 1.5527679180261796e-07 -9.953181886867287e-08 4.223925643871656
        children [
          DEF LEFTWHEELSHAPE Transform {
            translation 0.0425 0.02 0
            rotation 0 0 1 1.5707996938995747
            children [
              DEF WHEELSHAPE Shape {
                appearance DEF WHEELCOLOR PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry DEF WHEELBOTWHEELSHAPE Cylinder {
                  height 0.008
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "leftwheel"
        boundingObject USE LEFTWHEELSHAPE
        physics DEF WHEELPHYSICS Physics {
          density -1
          mass 0.05
        }
        linearVelocity -0.0017843345387787272 -4.89705178812759e-08 -0.05996962863065249
        angularVelocity -2.9985066486208916 8.46729332284997e-06 0.08923502703794887
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 17189.204526019075
        anchor 0 0.02 0
      }
      device [
        RotationalMotor {
          name "rightmotor"
          maxTorque 30
        }
      ]
      endPoint Solid {
        translation -6.301627750958109e-09 0.020393162233643417 0.019988412494409236
        rotation -0.9999999999999751 -1.639210844987034e-07 -1.5170278691187415e-07 1.5904744445141306
        children [
          DEF RIGHTWHEELSHAPE Transform {
            translation -0.0425 0.02 0
            rotation 0 0 1 1.5707996938995747
            children [
              USE WHEELSHAPE
            ]
          }
        ]
        name "rightwheel"
        boundingObject USE RIGHTWHEELSHAPE
        physics USE WHEELPHYSICS
        linearVelocity -0.001784342730127987 9.736987649983588e-08 -0.05996990365068649
        angularVelocity -2.998506647567561 7.690489505556106e-06 0.08923543047920894
      }
    }
    RangeFinder {
      translation 0 0.035 0.039
      rotation 0 1 0 3.141592653589793
      children [
        DEF RFSHAPE Transform {
          translation 0 0 -0.001
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance USE WHEELCOLOR
              geometry Cylinder {
                height 0.001
                radius 0.002
              }
            }
          ]
        }
      ]
      name "range finder"
      boundingObject USE RFSHAPE
      physics Physics {
      }
      fieldOfView 1.5
      width 300
      height 200
    }
  ]
  name "wheelbot"
  boundingObject USE WHEELBOTBODYSHAPE
  physics Physics {
    density -1
    mass 0.5
    centerOfMass [
      0 0.04 0
    ]
    inertiaMatrix [
      0.0009306968236259764 0.00030159289474462024 0.0009306968236259764
      0 0 0
    ]
  }
  controller "WheelBotController"
  linearVelocity -0.0017799608380197802 6.149098073264982e-06 -0.05996892064549328
  angularVelocity 0.00016008292635065805 0.0005031039466445132 -0.00019352040343289362
}
